newmiddleware/auth.js

```diff
@@ -0,0 +1,430 @@
+    )
+    auth.post("/v2/sites/:siteName/collections/:collectionName", this.verifyJwt)
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/subcollections/:subcollectionName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/collections/:collectionName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/collections/:collectionName/subcollections/:subcollectionName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/reorder",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/subcollections/:subcollectionName/reorder",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/move",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/collections/:collectionName/subcollections/:subcollectionName/move",
+      this.verifyJwt
+    )
+
+    // Resource Room
+    auth.get("/v2/sites/:siteName/resourceRoom", this.verifyJwt)
+    auth.post("/v2/sites/:siteName/resourceRoom", this.verifyJwt)
+    auth.get(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName",
+      this.verifyJwt
+    )
+
+    // Resource Categories
+    auth.get(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources/:resourceCategoryName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources/:resourceCategoryName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources/:resourceCategoryName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/resourceRoom/:resourceRoomName/resources/:resourceCategoryName/move",
+      this.verifyJwt
+    )
+
+    // Media directories
+    auth.get("/v2/sites/:siteName/media/:directoryName", this.verifyJwt)
+    auth.post("/v2/sites/:siteName/media", this.verifyJwt)
+    auth.post("/v2/sites/:siteName/media/:directoryName", this.verifyJwt)
+    auth.delete("/v2/sites/:siteName/media/:directoryName", this.verifyJwt)
+    auth.post("/v2/sites/:siteName/media/:directoryName/move", this.verifyJwt)
+
+    // Media files
+    auth.post("/v2/sites/:siteName/media/:directoryName/pages", this.verifyJwt)
+    auth.get(
+      "/v2/sites/:siteName/media/:directoryName/pages/:fileName",
+      this.verifyJwt
+    )
+    auth.post(
+      "/v2/sites/:siteName/media/:directoryName/pages/:fileName",
+      this.verifyJwt
+    )
+    auth.delete(
+      "/v2/sites/:siteName/media/:directoryName/pages/:fileName",
+      this.verifyJwt
```

Comment:
The 2 routers solution might be problematic, given that the auth middleware also checks and throws an error if the route requested is not one that we expect - I'll try a bit more to see if I can get it working, otherwise we might have to use the helper method instead

Reply:
As discussed offline, a second router is problematic because our current auth middleware implementation requires all routes to pass through it, and throws an error if the route is unrecognised - we've opted to use the helped method instead, implemented in https://github.com/isomerpages/isomercms-backend/pull/328/commits/84222d9ab39bc2e877a3d7b8503a10fec6004880